applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx

```diff
@@ -1,33 +1,169 @@
+  const [progress, setProgress] = useState(0)
+  const [remainingTime, setRemainingTime] = useState('55 min')
+
+  const pushErrorMessage = () => {
+    pushMessages([
+      {
+        content: (
+          <>
+            <Text>{t.onboarding.lastSteps.syncError}</Text>
+            <CtaButtonContainer>
+              <Button
+                variant='secondary'
+                onClick={() => dispatch(setOnboardingComplete(true))}
+              >
+                {t.common.verbs.continue}
+              </Button>
+            </CtaButtonContainer>
+          </>
+        ),
+        barFill: 0.875,
+        noSkip: true,
+        wait: 200,
+      },
+    ])
+  }
+
+  const startSync = async () => {
+    setSyncStarted(true)
+    pushMessages([
+      {
+        content: <Text>{t.onboarding.lastSteps.message2}</Text>,
+        barFill: 0.875,
+        noSkip: true,
+      },
+    ])
+    try {
+      await dispatch(baseNodeActions.startNode()).unwrap()
+    } catch (e) {
+      pushErrorMessage()
+    }
+  }
+
+  useEffect(() => {
+    if (syncStarted && updateMessageBoxSize && contentRef.current) {
+      updateMessageBoxSize()
+    }
+  }, [syncStarted])
+
+  const finishSyncing = () => {
+    dispatch(setOnboardingComplete(true))
+  }
+
+  /** MOCK WAITING FOR BASE NODE EVENT ABOUT SYNC PROGRESS */
+  const intervalRef = useRef<ReturnType<typeof setInterval> | undefined>()
+  const [intervalStarted, setIntervalStarted] = useState(false)
+  useEffect(() => {
+    if (syncStarted && !intervalStarted) {
+      let counter = 1
+      intervalRef.current = setInterval(async () => {
+        setProgress(counter * 10)
+        setRemainingTime(`${55 - counter * 5} min`)
+        if (counter === 6) {
+          pushErrorMessage()
+        }
+        if (counter > 10) {
+          finishSyncing()
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          clearInterval(intervalRef.current!)
+        }
+        counter++
+      }, 2000)
+      setIntervalStarted(true)
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return () => clearInterval(intervalRef.current!)
+  }, [syncStarted])
+  /** END OF MOCK WAITING FOR BASE NODE EVENT ABOUT SYNC PROGRESS */
+
+  return (
+    <FlexContent ref={contentRef}>
+      <Text as='span' type='defaultHeavy'>
+        {t.onboarding.lastSteps.message1} ✨💪
+      </Text>
+
+      {!syncStarted && (
+        <CtaButtonContainer>
+          <Button variant='primary' onClick={startSync}>
+            {t.onboarding.dockerInstall.startSyncBtn}
+          </Button>
+        </CtaButtonContainer>
+      )}
+
+      {syncStarted && (
+        <>
+          <Progress progress={progress} time={remainingTime} />
+          <CtaButtonContainer style={{ justifyContent: 'center' }}>
+            <Button
+              variant='secondary'
+              onClick={() => dispatch(setOnboardingComplete(true))}
```

Comment:
how does the application behave in this state?
if the user cancelled during sync or killed containers during sync - do we have any way of catching that state and fixing it?

Reply:
Originally, it was moving the user to the Home screen where Base Node and Tor were running. But on the second thought, I believe it should just close the application and stop containers. I changed this in last commit, so it closes the app. But waiting for the confirmation from Tola.

Reply:
ok, makes sense!